service.subtitles.wizdom/service.py

```diff
@@ -1,297 +1,345 @@
+				else:
+					imdb_id = "tt0"  # In order to show "No Subtitles Found" result => Doesn't recognize movie/episode
+	except Exception as err:
+		wlog('Caught Exception: error in imdb id: %s' % format(err))		
+		pass
+
+	if imdb_id[:2] == "tt":  # Simple IMDB_ID
+		searchByIMDB(imdb_id, item['season'], item['episode'], item['file_original_path'])
+	else:
+		# Search TV Show by Title
+		if item['season'] or item['episode']:
+			try:
+				imdb_id = searchTMDB("tv",quote(item['title']),0)
+				wlog("Search TV IMDB:%s [%s]" % (imdb_id, item['title']))
+				if imdb_id[:2] == "tt":
+					searchByIMDB(imdb_id, item['season'], item['episode'], item['file_original_path'])
+			except Exception as err:
+				wlog('Caught Exception: error in tv search: %s' % format(err))
+				pass
+		# Search Movie by Title+Year
+		else:
+			try:
+				imdb_id = searchTMDB("movie",query=item['title'], year=item['year'])
+				wlog("Search IMDB:%s" % imdb_id)
+				if not imdb_id[:2] == "tt":
+					imdb_id = searchTMDB("movie",query=item['title'], year=(int(item['year'])-1))
+					wlog("Search IMDB(2):%s" % imdb_id)
+				if imdb_id[:2] == "tt":
+					searchByIMDB(imdb_id, 0, 0, item['file_original_path'])
+			except Exception as err:
+				wlog('Caught Exception: error in movie search: %s' % format(err))
+				pass
+
+	# Search Local File
+	if not imdb_id:
+		ManualSearch(item['title'])
+	endOfDirectory(int(sys.argv[1]))
+	if myAddon.getSetting("Debug") == "true":
+		if imdb_id[:2] == "tt":
+			Dialog().ok("Debug "+myVersion, str(item), "imdb: "+str(imdb_id))
+		else:
+			Dialog().ok("Debug "+myVersion, str(item), "NO IDS")
 
 elif action == 'manualsearch':
-    searchstring = getParam("searchstring", params)
-    ManualSearch(searchstring)
-    endOfDirectory(int(sys.argv[1]))
+	searchstring = getParam("searchstring", params)
+	ManualSearch(searchstring)
+	endOfDirectory(int(sys.argv[1]))
 
 elif action == 'download':
-    id = getParam("id", params)
-    MyLog("Download ID:%s" % id)
-    subs = download(id)
-    for sub in subs:
-        listitem = ListItem(label=sub)
-        addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
-    endOfDirectory(int(sys.argv[1]))
+	id = getParam("id", params)
+	wlog("Download ID:%s" % id)
+	subs = download(id)
+	for sub in subs:
+		listitem = ListItem(label=sub)
+		addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
+	endOfDirectory(int(sys.argv[1]))
+	Ap = 0
+
+	# Upload AP
+	try:
+		if urlparse(Player().getPlayingFile()).hostname[-11:]=="tv4.live":
+			Ap = 1
+	except:
+		pass
+	
+	if Ap==1 and myAddon.getSetting("uploadAP") == "true":
+		try:
+			response = get("http://subs.vpnmate.com/webupload.php?status=1&imdb=%s&season=%s&episode=%s"%(getParam("imdb", params),getParam("season", params),getParam("episode", params)))
+			ap_object = loads(response.text)["result"]
+			if ap_object["lang"]["he"]==0:
+				xbmc.sleep(30*1000)
```

Comment:
sorry, if you want to sleep for 30 seconds you must use xbmc.Monitor:

`xbmc.Monitor().waitForAbort(30)`

Your current approach will hang kodi when exiting.